packages/core/lib/testing/testrunner.js

```diff
@@ -49,231 +50,164 @@ function TestRunner(options = {}) {
   this.TEST_TIMEOUT = (options.mocha && options.mocha.timeout) || 300000;
 }
 
-TestRunner.prototype.initialize = function(callback) {
-  var self = this;
-
-  var test_source = new TestSource(self.config);
+TestRunner.prototype.initialize = async function() {
+  debug("initializing");
+  let test_source = new TestSource(this.config);
   this.config.resolver = new TestResolver(
-    self.initial_resolver,
+    this.initial_resolver,
     test_source,
-    self.config.contracts_build_directory
+    this.config.contracts_build_directory
   );
 
-  var afterStateReset = function(err) {
-    if (err) return callback(err);
-
-    fs.readdir(self.config.contracts_build_directory, function(err, files) {
-      if (err) return callback(err);
-
-      files = _.filter(files, function(file) {
-        return path.extname(file) === ".json";
-      });
-
-      async.map(
-        files,
-        function(file, finished) {
-          fs.readFile(
-            path.join(self.config.contracts_build_directory, file),
-            "utf8",
-            finished
-          );
-        },
-        function(err, data) {
-          if (err) return callback(err);
-
-          var contracts = data.map(JSON.parse).map(json => contract(json));
-          var abis = _.flatMap(contracts, "abi");
+  if (this.first_snapshot) {
+    // // Make the initial deployment (full migration).
+    // await this.deploy();
+
+    try {
+      let initial_snapshot = await this.snapshot();
+      this.can_snapshot = true;
+      this.initial_snapshot = initial_snapshot;
+    } catch (_) {}
+    this.first_snapshot = false;
+  } else {
+    await this.resetState();
+  }
 
-          abis.map(function(abi) {
-            if (abi.type === "event") {
-              var signature =
-                abi.name + "(" + _.map(abi.inputs, "type").join(",") + ")";
-              self.known_events[web3Utils.sha3(signature)] = {
-                signature: signature,
-                abi_entry: abi
-              };
-            }
-          });
-          callback();
-        }
-      );
-    });
-  };
+  let files = await fs.readdir(this.config.contracts_build_directory);
 
-  if (self.first_snapshot) {
-    // // Make the initial deployment (full migration).
-    // self.deploy(function(err) {
-    //   if (err) return callback(err);
+  files = files.filter(file => path.extname(file) === ".json");
```

Comment:
How do you feel about just tacking this filter call on to the call to get the files above? It makes me feel weird when variables keep on being reassigned like this. If you don't want to then I won't fight you on it though.